src/integrations/ad-template/ad-template.service.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  Injectable,
+  Logger,
+  OnApplicationBootstrap,
+  OnModuleInit,
+} from '@nestjs/common';
+import * as mathjs from 'mathjs';
+import { Peripheral } from '@abandonware/noble';
+import { EntitiesService } from '../../entities/entities.service';
+import { ConfigService } from '../../config/config.service';
+import {
+  AdTemplateConfig,
+  AdTemplateDeviceOptions,
+  AdTemplateSensorOptions,
+} from './ad-template.config';
+import { DISTRIBUTED_DEVICE_ID } from '../home-assistant/home-assistant.const';
+import { makeId } from '../../util/id';
+import { Sensor } from '../../entities/sensor';
+import { EntityCustomization } from '../../entities/entity-customization.interface';
+import { SensorConfig } from '../home-assistant/sensor-config';
+import { BluetoothService } from '../bluetooth/bluetooth.service';
+
+interface templateScope {
```

Comment:
I'd use pascal case for interface names as well, so `TemplateScope`. Also applies to the other two interfaces below this one.